Code/Tools/AssetProcessor/native/ui/BuilderData.cpp

```diff
@@ -0,0 +1,154 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <native/ui/BuilderData.h>
+#include <utilities/AssetUtilEBusHelper.h>
+#include <native/ui/BuilderDataItem.h>
+#include <AzToolsFramework/AssetDatabase/AssetDatabaseConnection.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AssetProcessor
+{
+    BuilderData::BuilderData(AZStd::shared_ptr<AzToolsFramework::AssetDatabase::AssetDatabaseConnection> dbConnection, QObject* parent)
+        : m_dbConnection(dbConnection)
+        , QObject(parent)
+    {
+    }
+
+    //! This method runs when this model is initialized. It gets the list of builders, gets existing stats about analysis jobs and
+    //! processing jobs, and matches stats with builders and save them appropriately for future use.
+    void BuilderData::Reset()
+    {
+        BuilderInfoList builders;
+        AssetBuilderInfoBus::Broadcast(&AssetBuilderInfoBus::Events::GetAllBuildersInfo, builders);
+
+        m_root.reset(new BuilderDataItem(BuilderDataItem::ItemType::InvisibleRoot, "", 0, 0, nullptr));
+
+        m_allBuildersMetrics.reset(new BuilderDataItem(BuilderDataItem::ItemType::Builder, "All builders", 0, 0, m_root));
+        m_singleBuilderMetrics.clear();
+        m_builderGuidToIndex.clear();
+        m_builderNameToIndex.clear();
+        m_currentSelectedBuilderIndex = -1;
+
+        for (int i = 0; i < builders.size(); ++i)
+        {
+            m_singleBuilderMetrics.emplace_back(
+                new BuilderDataItem(BuilderDataItem::ItemType::Builder, builders[i].m_name, 0, 0, m_root));
+            m_builderGuidToIndex[builders[i].m_busId] = i;
+            m_builderNameToIndex[builders[i].m_name] = i;
+        }
+
+        // Analysis jobs stat
+        m_dbConnection->QueryStatLikeStatName(
+            "CreateJobs,%",
+            [this](AzToolsFramework::AssetDatabase::StatDatabaseEntry entry)
+            {
+                AZStd::vector<AZStd::string> tokens;
+                AZ::StringFunc::Tokenize(entry.m_statName, tokens, ',');
+                if (tokens.size() == 3) // CreateJobs,filePath,builderName
+                {
+                    const auto& sourceName = tokens[1];
+                    const auto& builderName = tokens[2];
+                    if (m_builderNameToIndex.contains(builderName))
+                    {
+                        m_singleBuilderMetrics[m_builderNameToIndex[builderName]]->UpdateOrInsertEntry(
+                            BuilderDataItem::JobType::CreateJob, sourceName, 1, entry.m_statValue);
+                        m_allBuildersMetrics->UpdateOrInsertEntry(
+                            BuilderDataItem::JobType::CreateJob, builderName + "," + sourceName, 1, entry.m_statValue);
+                    }
+                    else
+                    {
+                        AZ_Warning("AssetProcessor", false, "No builder found for an analysis job stat!!!\n");
```

Comment:
Can you make this warning more actionable? If someone encounters this, what's the impact and what action should they take?

Reply:
updated with actionable message. Can you check if my sentences make sense to you?